lib/prediction_class.py

```diff
@@ -79,11 +81,12 @@ def predict(self, input_file, allele, epitope_length, iedb_executable_path, iedb
 
             response = requests.post(self.url, data=data)
             retries = 0
-            while response.status_code == 500 and retries < iedb_retries:
-                time.sleep(60 * retries)
-                response = requests.post(self.url, data=data)
-                print("IEDB: Retry %s of %s" % (retries, iedb_retries))
+            while (response.status_code == 500 or response.status_code == 403) and retries < iedb_retries:
+                random.seed(uuid.uuid4().int)
```

Comment:
Shouldn't need to re-seed on every iteration of the loop, but it's also mostly harmless.